src/rules/code-diff.ts

```diff
@@ -0,0 +1,74 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2023 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import { HTMLElement, parse } from 'node-html-parser';
+import { groupBy, keyBy } from 'lodash';
+import { diffLines } from 'diff';
+const NUMBER_OF_EXAMPLES = 2;
+
+function replaceInDom(current: HTMLElement, code: string) {
+  const markedCode = new HTMLElement('pre', { class: 'code-difference-scrollable' }, '', current, null);
+  const flexDiv = new HTMLElement('div', { class: 'code-difference-container' }, '', current, null);
+  flexDiv.innerHTML = code;
+  markedCode.appendChild(flexDiv);
+  current.replaceWith(markedCode);
+}
+
+export function differentiateCode(compliant: string, nonCompliant: string) {
+  const hunks = diffLines(compliant, nonCompliant);
+
+  let nonCompliantCode = '';
+  let compliantCode = '';
+
+  hunks.forEach(hunk => {
+    if (!hunk.added && !hunk.removed) {
+      nonCompliantCode += `<div>${hunk.value}</div>`;
+      compliantCode += `<div>${hunk.value}</div>`;
+    }
+
+    if (hunk.added) {
+      compliantCode += `<div class='code-added'>${hunk.value}</div>`;
+    }
+
+    if (hunk.removed) {
+      nonCompliantCode += `<div class='code-removed'>${hunk.value}</div>`;
+    }
+  });
+  return [nonCompliantCode, compliantCode];
+}
+
+export function getExamplesFromDom(document) {
+  const pres = document.querySelectorAll(`pre[data-diff-id]`);
+
+  return (
+    Object.values(
+      groupBy(
+        pres.filter(e => e.getAttribute('data-diff-id') !== undefined),
+        e => e.getAttribute('data-diff-id')
```

Comment:
nit: Might be worth extracting the `'data-diff-id'` and `'data-diff-type'` attributes to a constant to highlight they are coming from the analyzers and we don't expect these to change